### PR TITLE
Fix Gradle release

### LINKED ISSUE #47: Release action incorrectly bumps for patch releases
https://github.com/bakdata/ci-templates/blob/main/.github/workflows/java-gradle-release.yaml#L83

1.0.0-SNAPSHOT will be bumped to 1.0.1 but should be bumped to 1.0.0

### LINKED ISSUE #48: Release will fail if CHANGELOG.md does not exist yet
https://github.com/bakdata/ci-templates/blob/8d8a3d9599e7d1f5b1197ddf93bed30c89481977/.github/workflows/java-gradle-release.yaml#L104 expects CHANGELOG.md to be part of the index already. Instead we need to do `git add CHANGELOG.md`

### PR DESCRIPTION
**Pull request by philipp94831**
Fixes #47
Fixes #48

### CODE REVIEW

File: .github/workflows/java-gradle-release.yaml [hunk 1]
```diff
@@ -41,9 +41,6 @@ on:
 
     # Map the workflow outputs to job outputs
     outputs:
-      old-version:
```
[the remark this answers is not in the document]
Comment (yannick-roeder) [reply .github/workflows/java-gradle-release.yaml:44]: Yes, I took inspiration from the bump-version action there, but didn't think about how versioning works in Gradle. I'm fine with removing it

File: .github/workflows/java-gradle-release.yaml [hunk 2]
```diff
@@ -80,9 +76,12 @@ jobs:
...
           old_version=$(grep "version=" gradle.properties | cut -d "=" -f2)
-          release_version=$(semver bump "${{ inputs.release-type }}" "${old_version}")
-          echo ::set-output "name=old-version::$old_version"
-          echo ::set-output "name=release-version::$release_version"
+          if [[ "${{ github.event.inputs.release-type }}" == "patch" ]]; then
+              release_version="${old_version%-*}"
+          else
+              release_version=$(semver bump "${{ github.event.inputs.release-type }}" "${old_version}")
+          fi
+          echo "release-version=$release_version" >> "$GITHUB_OUTPUT"
```
Comment (yannick-roeder) [inline .github/workflows/java-gradle-release.yaml:84]: What is this line for?

File: .github/workflows/java-gradle-release.yaml [hunk 3]
```diff
@@ -101,7 +100,8 @@ jobs:
 
       - name: Commit and push
         run: |
-          git commit -am "Changelog for version ${{ steps.evaluate-version.outputs.release-version }}" 
+          git add CHANGELOG.md
+          git commit -m "Changelog for version ${{ steps.evaluate-version.outputs.release-version }}"
```
Comment (yannick-roeder) [inline .github/workflows/java-gradle-release.yaml:104]: Does this really work when the changelog does not exist yet?

Comment (philipp94831) [reply .github/workflows/java-gradle-release.yaml:104]: Why shouldn't it? I add it to the index in the line before.

Comment (yannick-roeder) [reply .github/workflows/java-gradle-release.yaml:104]: But you cannot add files to the index if they don't exist right?

Comment (philipp94831) [reply .github/workflows/java-gradle-release.yaml:104]: No. But the file will exists after the previous steps

File: .github/workflows/java-gradle-release.yaml (hunk 2 above)
Comment (yannick-roeder) [reply .github/workflows/java-gradle-release.yaml:84]: Good to know

**Comment by yannick-roeder** [comment]
Closing in favor of https://github.com/bakdata/ci-templates/pull/86

**Pull request closed without merging**